yt_dlp/extractor/nhk.py

```diff
@@ -591,102 +592,198 @@ class NhkRadiruIE(InfoExtractor):
-        'skip': 'Expires on 2024-06-07',
+        'url': 'https://www.nhk.or.jp/radio/player/ondemand.html?p=J8792PY43V_14_4177121',
+        'skip': 'Expires on 2025-04-14',
         'info_dict': {
-            'id': '2834_01_4009298',
-            'title': 'まち☆キラ！開成町特集',
+            'id': 'J8792PY43V_14_4177121',
             'ext': 'm4a',
-            'release_date': '20240531',
-            'upload_date': '20240531',
-            'series': 'はま☆キラ！',
-            'thumbnail': 'https://www.nhk.or.jp/prog/img/2834/g2834.jpg',
-            'channel': 'NHK R1,FM',
-            'description': '',
-            'timestamp': 1717123800,
-            'uploader': 'NHK R1,FM',
-            'release_timestamp': 1717120800,
-            'series_id': '2834_01',
+            'title': '「白い蝶［ちょう］を探してみよう」いきもの☆いろいろ',
+            'description': '海野 和男（昆虫写真家）',
+            'uploader': 'NHK R1',
+            'channel': 'NHK R1',
+            'thumbnail': 'https://www.nhk.or.jp/radioondemand/json/J8792PY43V/img/corner/box_26_thumbnail.jpg',
+            'series': 'マイあさ！ いきもの☆いろいろ',
+            'series_id': 'J8792PY43V_14',
+            'timestamp': 1741985700,
+            'upload_date': '20250314',
+            'release_timestamp': 1741982400,
+            'release_date': '20250314',
         },
-        'expected_warnings': ['Failed to get extended metadata. API returned empty list.'],
+        'expected_warnings': ['Failed to download extended metadata: HTTP Error 404: Not Found'],
     }]
 
     _API_URL_TMPL = None
 
+    # the following few functions are ported from https://www.nhk.or.jp/radio/assets/js/timetable_detail_new.js
+
+    def _format_actlist(self, act_list):
+        role_groups = {}
+        for act in act_list:
+            role = act.get('role')
+            if role not in role_groups:
+                role_groups[role] = []
+            role_groups[role].append(act)
+
+        formatted_roles = []
+        for role, acts in role_groups.items():
+            for i, act in enumerate(acts):
+                res = f'【{role}】' if i == 0 and role is not None else ''
+                if title := act.get('title'):
+                    res += f'{title}…'
+                res += act.get('name')
+                formatted_roles.append(res)
+        return join_nonempty(*formatted_roles, delim='，')
+
+    def _fetch_artists(self, by_artist):
+        if not by_artist or len(by_artist) == 0:
+            return None
+        artists = []
+        for artist in by_artist:
+            res = ''
+
+            name = artist.get('name')
+            role = artist.get('role')
+            part = artist.get('part')
+
+            if role != '':
+                res += f'{role}…'
+            if part != '':
+                res += f'（{part}）'
+            res += name
+
+            if res != '':
+                artists.append(res)
+        if len(artists) == 0:
+            return None
+        return '、'.join(artists)
```

Comment:
Haven't tested this, but I think/hope this should work:
```suggestion
        artists = []
        for artist in traverse_obj(track, (key, ..., {dict})):
            if res := join_nonempty(*traverse_obj(track, (
                ...,
                ('role', filter, {'{}…'.format}),
                ('part', filter, {'（{}）'.format}),
                ('name', filter, {str}),
            )), delim=''):
                artists.append(res)

        return '、'.join(artists) or None
```

Reply:
it breaks things quite badly im afraid, in a way i don't really understand

closest i've got to working is this (in a separate test script):
```python
artist = {'name': 'ロイヤル・リバプール・フィルハーモニー管弦楽団', 'role': '', 'part': '管弦楽'}

print(artist)
print(traverse_obj(artist,
    ('role', filter, {'{}…'.format}, {str}),
    ('part', filter, {'（{}）'.format}, {str}),
    ('name', {str_or_none}),
))
```
it does work but only for one of them at once
ie currently it returns `（管弦楽）`
if i comment out the `part` line it returns `ロイヤル・リバプール・フィルハーモニー管弦楽団`
but i can't get it to return all of them

Reply:
welp that's why i should test
```py
        artists = []
        for artist in traverse_obj(track, (key, ..., {dict})):
            if res := join_nonempty(*traverse_obj(artist, ((
                ('role', filter, {'{}…'.format}),
                ('part', filter, {'（{}）'.format}),
                ('name', filter),
            ), {str})), delim=''):
                artists.append(res)

        return '、'.join(artists) or None
```

Reply:
:+1: it works